types/ext/language.d.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {TextSourceMap} from '../../ext/js/general/text-source-map.js';
+
+export type TextTransformationOption<T = unknown> = [
```

Comment:
Based on the other custom stuff we have, we probably will want to rename these so it's not ambiguous with the language transformer.

`TextPreprocessor` perhaps?